lib/libutee/include/user_ta_header.h

```diff
@@ -24,16 +24,39 @@
 	 */
 #define TA_FLAG_CONCURRENT		BIT32(8)
 	/*
-	 * Device enumeration is done in two stages by the normal world, first
-	 * before the tee-supplicant has started and then once more when the
-	 * tee-supplicant is started. The flags below control if the TA should
-	 * be reported in the first or second or case.
+	 * Device enumeration is initiated at multiple stages by the normal
+	 * world:
+	 * 1. First when the kernel driver has initialized
+	 * 2. When RPMB is available via inkernel RPMB routing
+	 * 3. When the tee-supplicant is started
+	 *
+	 * The flags below control at which stage a TA will be enumerated:
+	 * TA_FLAG_DEVICE_ENUM - at stage 1
+	 * TA_FLAG_DEVICE_ENUM_SEC_STORAGE -
+	 *      when secure storage is available, at stage 2 or 3 depending
+	 *      on whether TEE_STORAGE_PRIVATE is using RPMB FS
+	 *      (CFG_REE_FS=n CFG_RPMB_FS=y) or REE FS (CFG_REE_FS=y). The
+	 *      former utilizes in kernel RPMB routing, and the latter
+	 *      depends on tee-supplicant to access secure storage.
+	 *      TA_FLAG_DEVICE_ENUM_SUPP - at stage 3
+	 *
+	 * TA_FLAG_DEVICE_ENUM_SEC_STORAGE enumerates at stage 2 or 3
+	 * depending on whether TEE_STORAGE_PRIVATE is using RPMB FS
+	 * (CFG_REE_FS=n CFG_RPMB_FS=y) or REE FS (CFG_REE_FS=y). The
+	 * former utilizes in kernel RPMB routing, and the latter depends
+	 * on tee-supplicant to access secure storage.
+	 *
+	 * The TA is enumerated at stage 2 if
+	 * TA_FLAG_DEVICE_ENUM_SEC_STORAGE is set and TEE_STORAGE_PRIVATE
+	 * is using RPMB FS, or if it's using REE FS it will be
+	 * enumerated at stage 3.
```

Comment:
Consider the case where both `CFG_REE_FS` and `CFG_RPMB_FS` are `y`. A TA with `TA_FLAG_DEVICE_ENUM_SEC_STORAGE` which uses only `TEE_STORAGE_PRIVATE_RPMB` could in theory be enumerated at stage 2 but with the current code it will be enumerated at stage 3.
I have a feeling it can be a common case: enabling both FS to benefit from larger storage capacity, yet with specific TAs (such as fTPM) that should be enumerated before user space is available or need DoS protection and therefore would use pure RPMB storage via `TEE_STORAGE_PRIVATE_RPMB`. The TA flag needs to be more specific IMO.

Reply:
`TA_FLAG_DEVICE_ENUM_SEC_STORAGE` only concerns `TEE_STORAGE_PRIVATE`, also the storage used by fTPM. We'd need another flag for `TEE_STORAGE_PRIVATE_RPMB`.

Would `TA_FLAG_DEVICE_ENUM_TEE_STORAGE_PRIVATE` be better?
I guess we could add a `TA_FLAG_DEVICE_ENUM_TEE_STORAGE_PRIVATE_RPMB` too, but that would be without any user at the moment.

Reply:
> Would `TA_FLAG_DEVICE_ENUM_TEE_STORAGE_PRIVATE` be better?

Yes, I think so.